scientific_library/tvb/simulator/plot/head_plotter_3d.py

```diff
@@ -0,0 +1,103 @@
+#       The Virtual Brain: a simulator of primate brain network dynamics.
+#   Frontiers in Neuroinformatics (7:10. doi: 10.3389/fninf.2013.00010)
+#
+#
+
+
+import numpy as np
+import matplotlib.pyplot as plt
+from tvb.simulator.lab import *
+import ipywidgets as widgets
+from IPython.display import display
+
+class HeadPlotter3D:
+    def __init__(self):
+        pass
+
+    def display_source_sensor_geometry(self, meg_sensors = sensors.SensorsMEG.from_file(),
+                                        eeg_sensors = sensors.SensorsEEG.from_file(),
+                                        conn = connectivity.Connectivity.from_file(),
+                                        skin = surfaces.SkinAir.from_file()):
+        box_layout = widgets.Layout(border='solid 1px black', margin='3px 3px 3px 3px', padding='2px 2px 2px 2px')
+        
+        skin.configure()
+        
+        eeg_sensors.configure()
+
+        # CONTROLS
+        params = dict()
+        roi_checkbox = widgets.Checkbox(description="Show ROI Centers", value=False)
+        eeg_checkbox = widgets.Checkbox(description="Show EEG Sensors", value=False)
+        meg_checkbox = widgets.Checkbox(description="Show MEG Sensors", value=False)
+
+        control_box = widgets.HBox([roi_checkbox,eeg_checkbox,meg_checkbox], layout=box_layout)
+
+        params['ROI'] = roi_checkbox
+        params['EEG'] = eeg_checkbox
+        params['MEG'] = meg_checkbox        
+
+        def plot(**plot_params):
+            ax = plt.subplot(111, projection='3d')
+
+            # Plot boundary surface
+            x, y, z = skin.vertices.T
+            ax.plot_trisurf(x, y, z, triangles=skin.triangles, alpha=0.1, edgecolor='none')
+
+            if plot_params['ROI']:
+                # ROI centers as black circles
+                x, y, z = conn.centres.T
+                ax.plot(x, y, z, 'ko')
+            
+            if plot_params['EEG']:
+                # EEG sensors as blue x's
+                x, y, z = eeg_sensors.sensors_to_surface(skin).T
+                ax.plot(x, y, z, 'bx')
+
+            if plot_params['MEG']:
+                x, y, z = meg_sensors.locations.T
+                ax.plot(x, y, z, 'r+')
+        plt.figure()
+
+        out = widgets.interactive_output(plot, params)
+        display(control_box,out)
+
+    def display_surface_local_connectivity(self, ctx = cortex.Cortex.from_file()):
+
+        # configure cortical surface
+        loc_conn = local_connectivity.LocalConnectivity(cutoff=20.0, surface=ctx.region_mapping_data.surface)
+        loc_conn.equation.parameters['sigma'] = 10.0
+        loc_conn.equation.parameters['amp'] = 1.0
+        ctx.local_connectivity = loc_conn
+        ctx.coupling_strength = np.array([0.0115])
+        ctx.configure()
+
+        # plot 
+        plt.figure()
+        ax = plt.subplot(111, projection='3d')
+        x, y, z = ctx.vertices.T
+        ax.plot_trisurf(x, y, z, triangles=ctx.triangles, alpha=0.1, edgecolor='none')
```

Comment:
I agree with Paula. 
Plus, default params mutable are not recommended. 
Here is my proposal (unfortunately not fully tested, as my env got broken and I can't fix it right now)
```suggestion
    def display_surface_local_connectivity(self, ctx=None, loc_conn=None):
        # type: (cortex.Cortex, local_connectivity.LocalConnectivity) -> None
        """
        :param cortex: Optional cortex.Cortex instance. When none, we will try loading a default
        :param loc_conn: Optional. If None, and
        """
        # Start by configuring the cortical surface
        if ctx is None:
            ctx = cortex.Cortex.from_file()

        if ctx.local_connectivity is None:
            if loc_conn is None:
                loc_conn = local_connectivity.LocalConnectivity(cutoff=20.0, surface=ctx.region_mapping_data.surface)

            loc_conn.equation.parameters['sigma'] = 10.0
            loc_conn.equation.parameters['amp'] = 1.0
            ctx.local_connectivity = loc_conn

        ctx.coupling_strength = np.array([0.0115])
        ctx.configure()

        # plot 
        plt.figure()
        ax = plt.subplot(111, projection='3d')
        x, y, z = ctx.vertices.T
        ax.plot_trisurf(x, y, z, triangles=ctx.triangles, alpha=0.1, edgecolor='none')
```